pandas/core/frame.py

```diff
@@ -1399,11 +1399,13 @@ def to_dict(self, orient="dict", into=dict):
             )
         # GH16122
         into_c = com.standardize_mapping(into)
-        if orient.lower().startswith("d"):
+        if orient == "dict":
```

Comment:
Can you put in a line above 

```python
orient = orient.lower()
```

Reply:
but do we wanna accept a string like 'DICT' or 'Dict' as input? I am following the `to_json` `orient` option

Reply:
Actually I haven't thought about the ```orient``` option that is also available at ```DataFrame.to_json```, my thought was to avoid having an API change.

---

cc @jreback

Reply:
I see. I have also seen the `orient.lower()` inside the from_dict() method, which confused me. But the documentation states quite clearly which are the options for `orient`, which are all lower case, that's why I didn't think it would be a problem removing `.lower()`

Reply:
Agreed with @MomIsBestFriend. This would be an API change (though the other options are undocumented).

I would prefer to address and discuss this inconsistency with other methods in another issue

Reply:
Cool! I'll add it back then and open a new issue for the inconsistency :)